services/packagist/packagist-base.js

```diff
@@ -3,33 +3,62 @@
 const Joi = require('@hapi/joi')
 const { BaseJsonService } = require('..')
 
+const packageSchema = Joi.object()
+  .pattern(
+    /^/,
+    Joi.object({
+      version: Joi.string(),
+      require: Joi.object({
+        php: Joi.string(),
+      }),
+    }).required()
+  )
+  .required()
+
 const allVersionsSchema = Joi.object({
-  package: Joi.object({
-    versions: Joi.object()
-      .pattern(
-        /^/,
-        Joi.object({
-          version: Joi.string(),
-          require: Joi.object({
-            php: Joi.string(),
-          }),
-        })
-      )
-      .required(),
-  }).required(),
+  packages: Joi.object()
+    .pattern(/^/, packageSchema)
+    .required(),
 }).required()
-
 const keywords = ['PHP']
 
 class BasePackagistService extends BaseJsonService {
   async fetch({ user, repo, schema, server = 'https://packagist.org' }) {
```

Comment:
I _think_ I've got the gist of the difference between the JSON API and the Composer Metadata: TL;DR - JSON API has everything but is subject to 12 hour latency wheras Composer Metadata is always up to date but only has metadata about the package (no registry stats, etc.).

The docs you added to `fetchByJsonAPI` were very helpful (thanks!), but I still found myself bouncing between the two. Could we add docs to both functions to speak to their respective use cases? That way if a dev is working on one of the Packagist services they can get the doc for either function vs. having to always go to `fetchByJsonAPI`

Reply:
I thinks it's a very good idea.
Missing documentation is added :)

Reply:
Thanks!